coral/src/app/layout/main-navigation/MainNavigation.test.tsx

```diff
@@ -292,4 +222,37 @@ describe("MainNavigation.tsx", () => {
       });
     });
   });
+
+  describe("user can see their current team", () => {
+    beforeEach(() => {
+      customRender(<MainNavigation />, {
+        memoryRouter: true,
+        queryClient: true,
+      });
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      useQuerySpy.mockReturnValue({ data: "Team-name", isLoading: true });
+    });
+
+    afterEach(() => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      useQuerySpy.mockReturnValue({ data: "Team-name", isLoading: false });
+      cleanup();
+    });
+
+    it("renders loading state", () => {
+      const teamLabel = screen.getByText("Team");
+      const teamName = screen.getByText("Fetching team..."); // -> Fails because teamName is "Team-name"
```

Comment:
I had to wrap these in a `await waitFor(...)` to make it properly wait until the component  query has been called.